src/pages/Share/ShareRootPage.tsx

```diff
@@ -33,6 +36,25 @@ function showErrorAlert(title: string, message: string) {
 }
 
 function ShareRootPage() {
+    const [currentAttachment] = useOnyx(ONYXKEYS.SHARE_TEMP_FILE, {canBeMissing: true});
+
+    const {validateFiles} = useFilesValidation(addValidatedShareFile);
+    const isTextShared = currentAttachment?.mimeType === 'txt';
+
+    const validateFilesWrapper = (file: ShareTempFile) => {
+        if (!file || isTextShared) {
+            return;
+        }
+
+        validateFiles([
+            {
+                name: file.id,
+                uri: file.content,
+                type: file.mimeType,
+            },
+        ]);
+    };
```

Comment:
We're using validatedFile only when sharing HEIC files based on this logic

```ts
const shouldUsePreValidatedFile = currentAttachment?.mimeType === CONST.SHARE_FILE_MIMETYPE.HEIC;
```

Let's validate the file only for this type to get rid of unnecessary calls

Reply:
NAB, Also I'm not a huge fan of this wrapper, as we're using this function only once

Reply:
It seems to me that we can do it this way, but a comment should be added stating that the flags on the root and details pages must be consistent.